legate/tester/test_system.py

```diff
@@ -117,13 +129,25 @@ def run(
         full_env = dict(os.environ)
         full_env.update(env)
 
-        proc = stdlib_run(
-            cmd, cwd=cwd, env=full_env, stdout=PIPE, stderr=STDOUT, text=True
-        )
-
+        t0 = datetime.now()
+        try:
+            proc = stdlib_run(
+                cmd,
+                cwd=cwd,
+                env=full_env,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+                timeout=timeout,
+            )
+        except TimeoutExpired:
+            return ProcessResult(invocation, test_file, timeout=True)
```

Comment:
Does it make sense to save the timed-out test's output (and present it on the final output), in case it's useful to find out what specific function hung?

Reply:
It's not available, AFAIK
```
In [10]: try:
    ...:     run(["yes"], timeout=1)
    ...: except Exception as e:
    ...:     print(e.stderr)

y
y
y
y
y
None
```
same for `stdout` on the exception.

Reply:
Unfortunately that won't work with the way we currently run processes:
```
In [3]: try:
   ...:     proc = run(["yes"], stdout=PIPE, stderr=STDOUT, text=True, timeout=1)
   ...: except Exception as e:
   ...:     print(proc.stdout)
   ...: 

TimeoutExpired: Command '['yes']' timed out after 1 seconds

During handling of the above exception, another exception occurred:

NameError                                 Traceback (most recent call last)
Cell In[3], line 4
      2     proc = run(["yes"], stdout=PIPE, stderr=STDOUT, text=True, timeout=1)
      3 except Exception as e:
----> 4     print(proc.stdout)

NameError: name 'proc' is not defined
```
Maybe it's possible to re-work things to use `capture_output=True` instead of pipes but maybe that could be investigated as a follow-on.

Reply:
Yup, an investigation for another day